Name ecommerce.items parameter names in purchase event GA4-compliant

wmdkfhartung opened the pull request:
Currently some of the items parameter names in the purchase event are not named GA4 compliant. (https://github.com/d3datadevelopment/GoogleAnalytics4/blob/master/Application/views/blocks/checkout_s5.tpl)

So they are called "id, name, variant", although they should be called "item_id, item_name, item_variant". (See https://developers.google.com/analytics/devguides/collection/ga4/set-up-ecommerce?hl=de#add-event)

In the view_item, add_to_cart and view_cart event they are named correctly.

d3datadev commented:
In coordination with the initial author of the module, we are currently continuing the development. Therefore, please feel free to use our fork (https://github.com/d3datadevelopment/GoogleAnalytics4/) for your collaboration.

The changes can be returned to the parent repository at any time. However, I would not expect this.

MaxBuhe01 committed:
bump version and changelog to 2.9.0

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_2.x'

# Conflicts:
#	CHANGELOG.md

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_1.x'

# Conflicts:
#	Application/views/admin/de/module_options.php
#	Application/views/ga4/view_item_list.tpl
#	CHANGELOG.md
#	Modules/Core/ViewConfig.php
#	composer.json
#	metadata.php

MaxBuhe01 committed:
[bug-fix] non-category having articles throw exception

deny method-call on no category

MaxBuhe01 committed:
bump changelog && metadata to 2.9.1

MaxBuhe01 committed:
[fix-bug] wrong category method used for check

MaxBuhe01 committed:
bump changelog && version to 2.9.2

MaxBuhe01 committed:
bump changelog && version to 2.10.0

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_2.x'

MaxBuhe01 committed:
add remove_from_cart

MaxBuhe01 committed:
add missing param to check if isRemoveFromCart

MaxBuhe01 committed:
bump changelog && version to 2.10.1

MaxBuhe01 committed:
[working-state] rebuilding meta structure

gaertnermarkus committed:
extend Manufacturer

gaertnermarkus committed:
extend Manufacturer

MaxBuhe01 committed:
correct naming in view_search_result

MaxBuhe01 committed:
add deug_mode switch

MaxBuhe01 committed:
general structure rework, debug_mode setter

MaxBuhe01 committed:
add new method for debug-check

MaxBuhe01 committed:
add 'remove_from_cart' event

MaxBuhe01 committed:
extended docs

MaxBuhe01 committed:
add gitignore

MaxBuhe01 committed:
add CountryObjectGetter to ThankYouController

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/dev_1.x_manufacturer' into dev_1.x

# Conflicts:
#	metadata.php

MaxBuhe01 committed:
extends changelog

MaxBuhe01 committed:
[fixed] metadata class entry

MaxBuhe01 committed:
[changed] 'an article has been removed from basket'-check

MaxBuhe01 committed:
[changed] 'an article has been removed from basket'-check

MaxBuhe01 committed:
bump metadata && changelog

MaxBuhe01 committed:
add cookiebot essentials

MaxBuhe01 committed:
add cookiebot essentials

MaxBuhe01 committed:
bump metadata && changelog

MaxBuhe01 committed:
bump metadata && changelog

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_2.x'

MaxBuhe01 committed:
adjust Doku with further infos new cookieid -> Control parameter

MaxBuhe01 committed:
adjust Doku with further infos new cookieid -> Control parameter

MaxBuhe01 committed:
switch name of control-parameter; switch consentmanager-script-adjustments

# Conflicts:
#	Modules/Core/ViewConfig.php

MaxBuhe01 committed:
switch name of control-parameter; switch consentmanager-script-adjustments

MaxBuhe01 committed:
switch param-name

MaxBuhe01 committed:
switch control-param name

MaxBuhe01 committed:
adjust set-param check

MaxBuhe01 committed:
adjust set-param check

MaxBuhe01 committed:
cleaning; adjust not set param handling

MaxBuhe01 committed:
bump metadata and readme

MaxBuhe01 committed:
bump metadata && changelog

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_1.x'

# Conflicts:
#	Application/views/admin/de/module_options.php
#	Application/views/ga4/view_item_list.tpl
#	CHANGELOG.md
#	Docs/README.md
#	IntelliSenseHelper.php
#	Modules/Core/ViewConfig.php
#	metadata.php

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_2.x'

# Conflicts:
#	Modules/Core/ViewConfig.php

MaxBuhe01 committed:
[Added] add_to_cart to page/list/list.tpl

MaxBuhe01 committed:
[Changed] process-list of add_to_cart; [Added] BasketComponent extensions to controll the trigger-behaviour of add_to_cart, ArticleDetailsController extension for add_to_Cart i view_item

add_to_cart no longer is a block extension, changed to an indiv tpl-file

MaxBuhe01 committed:
[Added] add_to_cart to page/list/list.tpl, BasketComponent methods for trigger-behaviour add_to_cart

MaxBuhe01 committed:
[Added] add_to_cart to recommlist, wishlist, noticelist, Class-Extensions for corresponding templates

MaxBuhe01 committed:
[Added] add_to_cart to start-template

MaxBuhe01 committed:
[Changed] capture implementation

reworked to a safer path-through to get the data

MaxBuhe01 committed:
[Added] phpstan + phpstan file

MaxBuhe01 committed:
[Added] missing IntelliSenseHelper entry for Manufacturer

MaxBuhe01 committed:
[Changed] template structure, template-structure information

all block extensions now only include the template instead of having the extension included immediately (dedicated event-templates within views/event); information about the structure changed to the correct paths

MaxBuhe01 committed:
[Fixed] error on array-count at zero entries

MaxBuhe01 committed:
bump changelog && version to 2.13.1

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_2.x'

MaxBuhe01 committed:
[Added] search controller extension (add_to_cart)

MaxBuhe01 committed:
[Changed] phpStan - set level to 3, clean files by level

MaxBuhe01 committed:
[Changed] phpStan - set level to 4, clear files by level

Also changed method, to get the category for items (products)

MaxBuhe01 committed:
[Working] set PHPStan to lvl 5

MaxBuhe01 committed:
[Added] PHPStan lvl 5, error-ignoring; [Fixed] user-Getter evaluation

MaxBuhe01 committed:
[Bump] metadata && changelog

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_1.x' into dev_2.x

MaxBuhe01 committed:
bump changelog && version to 2.14.0

MaxBuhe01 committed:
[Changed] phpstan.neon

MaxBuhe01 committed:
[Changed] phpstan.neon

MaxBuhe01 committed:
[Added] extended Doks

MaxBuhe01 committed:
[Fix] missing Component-ArticleDetails extension; missing ManufacturerList-Extension

MaxBuhe01 committed:
[Fix] missing Component-ArticleDetails extension; missing ManufacturerList-Extension

MaxBuhe01 committed:
[Bump] Changelog && Metadata

MaxBuhe01 committed:
[Bump] Metadata && Changelog

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_1.x'

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_2.x'

MaxBuhe01 committed:
[OST#929760][Added] serverside tagging usability

*missing any test, yet!

MaxBuhe01 committed:
[OST#929760][Added] multilang texts for desc

Test: OK

MaxBuhe01 committed:
[Bump] Changelog && Metadata

MaxBuhe01 committed:
[OST830498][Changed] metadataid centralized

MaxBuhe01 committed:
[Fixed] potential error

if no basket-component is given, error occurs

MaxBuhe01 committed:
[OST830498][Added] consent-mode standard values

MaxBuhe01 committed:
[OST830498][Added] multilang ident

MaxBuhe01 committed:
[OST830498][Fixed] dataLayer overwriting

MaxBuhe01 committed:
[OST830498][Fixed] comment-error

Syntax-Error duo to the OXID-strip that made the inline-comment cut off "})" of the code

MaxBuhe01 committed:
[OST830498][Removed] ( for now ) unnecessary set

MaxBuhe01 committed:
[Changed] styling

MaxBuhe01 committed:
[Bump] Metadata && Changelog

MaxBuhe01 committed:
[Fixed] getGtmContainerId gave false/ null

on second call method gave false

MaxBuhe01 committed:
[Bump] Metadata && Changelog

MaxBuhe01 committed:
[Bump] changelog

MaxBuhe01 committed:
[Bump] metadata && changelog

MaxBuhe01 committed:
[Bump] changelog

MaxBuhe01 committed:
[Added] trait for generalized method

MaxBuhe01 committed:
[Added] unreleased branch comparison to chnagelog

MaxBuhe01 committed:
[Changed] declare module settings consistency

MaxBuhe01 committed:
[Added] new Admin-UI

MaxBuhe01 committed:
[Changed] configParamGetter

MaxBuhe01 committed:
[Removed] deprecated and obsolete module-settings

MaxBuhe01 committed:
[Changed] clear dead code

MaxBuhe01 committed:
[Fixed] false class-property declaration

MaxBuhe01 committed:
[Added] additional check for properly activated module

MaxBuhe01 committed:
[Changed] template code upgrade

change generic-html-tag id and controll names to specified names

MaxBuhe01 committed:
[Added] admin headitem alternative

MaxBuhe01 committed:
[OST497528][Added] usercentrips step/ option 2

detailed information about the new addings check docs: https://docs.usercentrics.com/#/consent-mode

MaxBuhe01 committed:
[Added] new CMP individualisation to own Model

this is the first step into manifesting "big" CMP's into own Models that require strong individualism partially

MaxBuhe01 committed:
[Changed] clear code

MaxBuhe01 committed:
[Bump] metadata for testing

MaxBuhe01 committed:
[Changed] lang text

MaxBuhe01 committed:
[Changed] amount of save-buttons

MaxBuhe01 committed:
[Added] Help text

multilang + template entries

MaxBuhe01 committed:
[Added] Container-ID explanation to docs

MaxBuhe01 committed:
[Added] extended translations/ explanations

MaxBuhe01 committed:
[Changed] Usercentrics-Class constants and adjust glob-code to it

MaxBuhe01 committed:
[Changed] way of manager-classification in tpl

MaxBuhe01 committed:
[Added] auto safe on first approach ga4 + cmp-mod

MaxBuhe01 committed:
[Changed] clear code and adjust Naming

MaxBuhe01 committed:
[Bump] Metadata && Changelog

MaxBuhe01 committed:
[Removed] false xml excludation

MaxBuhe01 committed:
[Added] extended changelog entries

MaxBuhe01 committed:
[Added] onActivate/ onDeactivate

MaxBuhe01 committed:
[Bump] Changelog

MaxBuhe01 committed:
[Fixed] wrong to-save param

MaxBuhe01 committed:
[Bump] Changelog

MaxBuhe01 committed:
[Fixed] missing return statements, on null

MaxBuhe01 committed:
[Bump] Metadata && Changelog

MaxBuhe01 committed:
[Fixed] overwriting existing settings

accidentally the existing settings were overwritten

MaxBuhe01 committed:
[Fixed] overwriting existing settings

accidentally the existing settings were overwritten

MaxBuhe01 committed:
Merge remote-tracking branch 'origin/dev_2.x_fix_onactivate' into dev_2.x_fix_onactivate

# Conflicts:
#	Setup/Actions.php

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/dev_2.x_fix_onactivate' into rel_2.x

MaxBuhe01 committed:
[Bump] Metadata && Changelog

MaxBuhe01 committed:
[Change] classify composer requirements more strictly

classify PHP and OXID-CE-Version

MaxBuhe01 committed:
[Fixed] missing class extension

Activating the Module on CLI happens so early, there's no ModuleChain available yet, therefor no extension, therefor no additional methods

MaxBuhe01 committed:
[Fixed] typo

MaxBuhe01 committed:
[Fixed] typo

MaxBuhe01 committed:
Merge remote-tracking branch 'origin/dev_2.x_12787_bug_fixing' into dev_2.x_12787_bug_fixing

gaertnermarkus committed:
Prevent "OXID Logger.ERROR: D3\GoogleAnalytics4\Modules\Application\Controller\d3GtmStartController is not an instance of OxidEsales\Eshop\Application\Component\Widget\WidgetController"

gaertnermarkus committed:
Erweiterung verschieben

gaertnermarkus committed:
Erweiterung verschieben

gaertnermarkus committed:
render() ergänzen

MaxBuhe01 committed:
[Bump] Metadata && Changelog

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_2.x'

MaxBuhe01 committed:
[Changed] clean code

MaxBuhe01 committed:
[Added] new Event "begin_checkout"

- new block extension
- new template for the event-data for GA
- metadata adjustments for it
- new version ( 2.20.0 )

MaxBuhe01 committed:
[Added] new Event "add_payment_info"

- add Payment-Model extension for own small method
- new block extension
- new template for the event-data for GA
- metadata adjustments for it

MaxBuhe01 committed:
[Bump] Metadata

gaertnermarkus committed:
Methode umbenannt

MaxBuhe01 committed:
[Fixed] currency in begin_checkout

MaxBuhe01 committed:
Bump metadata && changelog

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/dev_2.x_d3GtmStartController' into rel_2.x

MaxBuhe01 committed:
[Bump] Metadata && Changelog

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_2.x'

MaxBuhe01 committed:
[Removed] unnecessary "none" entry for CMP-choosability

MaxBuhe01 committed:
[Fixed] Consentmanager doesn't get delivered

false value given for important decision-making. Since the value was false, we got "false" as return for D3blShowGtmScript() which made consentmanager not send the data to GTM

MaxBuhe01 committed:
[Changed] add "internal_names" to constant

MaxBuhe01 committed:
[Bump] Metadata && Changelog

MaxBuhe01 committed:
[Changed] OXID7 metadata adjustments

MaxBuhe01 committed:
[Changed] OXID 7 file adjustments

MaxBuhe01 committed:
[Changed] composer reqs

MaxBuhe01 committed:
Merge branch 'dev_3.x_OXID7' of https://git.d3data.de/D3Public/GoogleAnalytics4 into dev_3.x_OXID7

MaxBuhe01 committed:
[Fixed] template paths

MaxBuhe01 committed:
[Added] module-activated check

since its a private method in OXID7

gaertnermarkus committed:
use Constants ergänzt

SeifertDaniel committed:
don't submit item prices, if user doesn't have "show price" right

# Conflicts:
#	Application/views/ga4/remove_from_cart.tpl
#	Application/views/ga4/view_search_result.tpl

SeifertDaniel committed:
don't submit item prices, if user doesn't have "show price" right

MaxBuhe01 committed:
[Changed] code cleanup

php cs fixer

MaxBuhe01 committed:
[Fixed] view_cart template register

MaxBuhe01 committed:
[Changed] version constraint

MaxBuhe01 committed:
[Fixed] bash-hints readme; [Added] changelog entries

MaxBuhe01 committed:
[OST614872][Added] method for additional globals

MaxBuhe01 committed:
[Bump] Metadata && Changelog

MaxBuhe01 committed:
[Fixed] false metadata view_cart entry

MaxBuhe01 committed:
[Fixed] missing menu-translation

MaxBuhe01 committed:
[Changed] composer.json meta-reqs

MaxBuhe01 committed:
[Bump] Metadata & Changelog

MaxBuhe01 committed:
[Bump] metadata + Changelog

MaxBuhe01 committed:
[Changed] clear code

smxsm committed:
MGH-424 easycredit fix: cast oxdesc to string, may be object

smxsm committed:
MGH-424 easycredit fix: cast oxdesc to string, may be object

MaxBuhe01 committed:
[Changed] remove save-logic from Getter-Method

MaxBuhe01 committed:
[Fixed] over-engineered Getter-methods

- the methods would semi-automatic search and set/ save params
- I removed those and set it back to its base functionality:
-- saving happens ONLY at the save() function if triggered by the user in the admin-frontend
-- only one Getter who checks for a value in the DB ? give the value : no value given yet

MaxBuhe01 committed:
[Removed] over-engineered Getter-methods

- the methods would semi-automatic search and set/ save params
- I removed those and set it back to its base functionality:
-- saving happens ONLY at the save() function if triggered by the user in the admin-frontend
-- only one Getter who checks for a value in the DB ? give the value : no value given yet

MaxBuhe01 committed:
[Bump] Changelog && Metadata

MaxBuhe01 committed:
[Bump] Changelog && Metadata

MaxBuhe01 committed:
Fix changelog

MaxBuhe01 committed:
[Fixed] changelog

MaxBuhe01 committed:
[Bump] Changelog

FlorianPalme committed:
added option to add variants in view_item event

FlorianPalme committed:
added option to add variants in view_item event

smxsm committed:
Add missing semikolon to prevent javascript errors

smxsm committed:
Add missing semikolon to prevent javascript errors

smxsm committed:
Add item_oxid parameter to all event templates

smxsm committed:
Add item_oxid parameter to all event templates

MaxBuhe01 committed:
[Added] twig files ( unfixed )

MaxBuhe01 committed:
- Change item_category_X to item_categoryX etc.

- add item_brand to all events
- add option to use real category titles instead of url parts for item_categoryX
- add option to replace chars in item_categoryX values

thanks to Stefan Moisees@beffy@proudcommerce.com

MaxBuhe01 committed:
- Change item_category_X to item_categoryX etc.

- add item_brand to all events
- add option to use real category titles instead of url parts for item_categoryX
- add option to replace chars in item_categoryX values

thanks to Stefan Moisees@beffy@proudcommerce.com

MaxBuhe01 committed:
[Changed] moving new setting-sections to corresponding area

MaxBuhe01 committed:
[Changed] moving new setting-sections to corresponding area

MaxBuhe01 committed:
[Fixed] no input error

if activated but no chars are given, it'll cut out the whole category-string
-> added char-check

MaxBuhe01 committed:
[Fixed] no input error

if activated but no chars are given, it'll cut out the whole category-string
-> added char-check

MaxBuhe01 committed:
[Changed] explanation text

MaxBuhe01 committed:
[Changed] explanation text

MaxBuhe01 committed:
[Bump] changelog && Metadata

MaxBuhe01 committed:
[Fixed] irritating admin list module group-name

MaxBuhe01 committed:
[Bump] Changelog

MaxBuhe01 committed:
[Changed] move item_variants to own event-parameter section

MaxBuhe01 committed:
[Fixed] composer correlation

MaxBuhe01 committed:
[Added] .htaccess for assets-folder

MaxBuhe01 committed:
[Fixed] smarty to twig converted templates

MaxBuhe01 committed:
[Changed] reconstructed to fit extending schemata

MaxBuhe01 committed:
[Changed] implement twig extensions

MaxBuhe01 committed:
[Changed] file-paths of included files

MaxBuhe01 committed:
[Changed] lang files for OX72

MaxBuhe01 committed:
reverting accidental deletion

MaxBuhe01 committed:
[Fixed] lang files admin-list

MaxBuhe01 committed:
[Reverted] file-content

MaxBuhe01 committed:
[Fixed] wrong twig-modifier

MaxBuhe01 committed:
[Fixed] wrong twig-modifier

MaxBuhe01 committed:
[Fixed] wrong twig-modifier

MaxBuhe01 committed:
[Reverted] original twig-content

MaxBuhe01 committed:
[Fixed] missing |raw modifier

MaxBuhe01 committed:
[Fixed] twig keywords

MaxBuhe01 committed:
[Fixed] twig

MaxBuhe01 committed:
Merge remote-tracking branch 'origin/dev_3.x__OX7_twig' into dev_3.x__OX7_twig

# Conflicts:
#	views/twig/admin/d3ga4uiheaditem.html.twig
#	views/twig/admin/d3ga4uimain.html.twig

MaxBuhe01 committed:
[Fixed] wrong html-id input

MaxBuhe01 committed:
[Removed] invalid files

MaxBuhe01 committed:
[Added] IDE-Support script-tags and rearranging content

MaxBuhe01 committed:
[Fixed] missing currency-name

MaxBuhe01 committed:
[Fixed] missing currency-name

MaxBuhe01 committed:
[Fixed] missing checkout/user extension

making sure that begin_checkout is always rolled out

MaxBuhe01 committed:
[Added] band_name + rawFieldData

MaxBuhe01 committed:
[Added] option to add variants in view_item event

by  Florian Palme <info@florian-palme.de>

MaxBuhe01 committed:
[Fixed] lang file

MaxBuhe01 committed:
[Changed] move item_variants to own event-parameter section

MaxBuhe01 committed:
[Changed] move new settings in ga4-mod-ui to corresponding area

MaxBuhe01 committed:
[Fixed] Param-Call and missing comma

MaxBuhe01 committed:
[Fixed] output-param name

MaxBuhe01 committed:
[Changed] save settings standard-way

MaxBuhe01 committed:
[Added] Todo's

MaxBuhe01 committed:
[working-state][Changed] save/ Get methods

MaxBuhe01 committed:
[working-state][Changed] save/ Get methods

MaxBuhe01 committed:
[Fixed] commenting out add_to_cart

since it doesn't work properly currently and requires a greater fixup

MaxBuhe01 committed:
[Fixed] false object for manufacturer ( purchase.twig )

MaxBuhe01 committed:
[Changed] data getter to state-of-the-art

and removed unnecessary code/ try-outs

MaxBuhe01 committed:
[Changed] Metadata-Version

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_2.x' into dev_3.x__OX7_twig

MaxBuhe01 committed:
[Fixed] namespace inclusion, admin-UI list title, potential error on whitespaces in container ID

MaxBuhe01 committed:
[Bump] Changelog && metadata

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_2.x'

MaxBuhe01 committed:
Merge remote-tracking branch 'remotes/origin/rel_3.x'
